### DISCUSSION TITLE
Permissions to limit user list for 'Owner', 'Created by' dropdown - Filters and Search

### DISCUSSION THREAD

**Discussion by etadelta222** [apache/superset#18299] [Q&A / Help]
Our organization has external users who have access to dashboards specifically designed for them. We are running an older version of superset and upgrading to the 1.1.0. In the new version the UI has filter and search capabilities on the dashboard page (referenced [here](https://docs.preset.io/docs/how-to-find-charts-and-dashboards)). 

Even though user accessing the dashboard only has access to their dashboard, the dropdown list for 'Owners' and 'Created By' display ALL users. This is an issue since we don't want external users to be able to see other users.

![image](https://user-images.githubusercontent.com/19858818/121082751-8e432700-c7ac-11eb-9120-51cb903c9be9.png)

Since we are restricting the user access to specific dashboards using roles and permissions, a new permission to hide the filters altogether would be ideal.

Current options:
The values are being populated by calling the /dashboard/related/[column_name] API. We would have to pass in query parameter to only show current user OR based on user/role only show 'All' as default value OR hide 'Owners' and 'Created By'. 

https://github.com/apache/superset/blob/74473e2ad1975150a6dc578d67aae05de5ba4b02/superset-frontend/src/views/CRUD/dashboard/DashboardList.tsx#L386

I've asked about this in the Apache Superset Slack workspace and have not had any luck. Please let me know if I can provide any additional information.

**Comment by nytai** [comment]
the issue here is that permissions are not at the row level. So a user either has a permission to fetch a list of users or not. You might be able to achieve what you need using row level security

**Comment by etadelta222** [comment]
Thanks for the suggestion @nytai. I actually did try that approach but it did not work. The user information is stored in the ab_user table and I'm not able to access it for RLS filter. I'm logged in as admin.

![image](https://user-images.githubusercontent.com/19858818/121085166-88027a00-c7af-11eb-9a01-f4b641ff83c2.png)

**Comment by etadelta222** [comment]
Just wanted to check if 1.2 addresses this.

**Comment by russmatney** [comment]
This behavior is indeed the same in 1.2 - the dropdowns here expose all users to any other user that visits the dashboards list. This lack of control is unfortunate, as it prevents this feature (and features like it https://github.com/apache/superset/issues/15066) from being usable in a multi-tenant context (without forking/workarounds), where some groups of users and datasources should not be exposed to each other.

**Comment by etadelta222** [comment]
Thanks @russmatney. I was hoping it was addressed and the upgrade will be less painful. I know there were front end changes from 1.1.0 to 1.2 so I won't be able to just overwrite files that I had to update in 1.1.0. :/

**Comment by mijbel-raedata** [comment]
Hi @etadelta222 Have you figured this out?

**Reply by mayank-lalwani** [reply]
Hi, has anyone figured this out yet?

**Reply by mijbel-raedata** [reply]
You can either edit the source code or use as is. Couldn't find another solution

**Reply by mayank-lalwani** [reply]
in which file do we need to make the changes in the source code?

**Reply by etadelta222** [reply]
MenuRight.tsx

**Reply by etadelta222** [reply]
See Tai's response in slack: https://apache-superset.slack.com/archives/CCKHMGRRB/p1663179112817529

**Comment by bcook-konza** [comment]
I would like resurrect this thread if possible - we are running into the same issue. Has there been any progress on supporting this since 2021 or would creating our own fork be the only option as of now?

**Comment by marre** [comment]
I think you are looking for the EXTRA_RELATED_QUERY_FILTERS option.

Have a look at the summary of this PR https://github.com/apache/superset/pull/29287 for an example on how to configure this.
